apollo-gradle-plugin/src/main/groovy/com/apollostack/android/gradle/ApolloIRgenTask.groovy

```diff
@@ -0,0 +1,60 @@
+package com.apollostack.android.gradle
+
+import com.google.common.collect.Sets
+import com.moowork.gradle.node.task.NodeTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.util.PatternSet
+
+public class ApolloIRgenTask extends NodeTask {
+  private static final String APOLLO_CODEGEN = "node_modules/apollo-codegen/lib/cli.js"
+  static final String NAME = "generate%sApolloIR"
+  List<ApolloExtension> config
+  String variant
+
+  @InputFiles
+  Collection<File> getInputFiles() {
+    Set<File> inputFiles = Sets.newHashSet()
+    config.each { ext ->
+      ext.getFiles().asMap().entrySet().each { entry ->
+        entry.value.each { file ->
+          inputFiles.add(new File(entry.key, file))
+        }
+      }
+    }
+    return inputFiles
+  }
+
+  @Override
+  public void exec() {
```

Comment:
> you need to be able to generate the java interfaces in the same directory hierarchy as the original .graphql files?

yup, this.

> do we need to check for that though, or just let apollo-codegen fail for us?

I'd say let apollo-codgen do it. The error thrown is pretty descriptive.

```
There can only be one operation named "HeroName".
There can only be one operation named "HeroName".
There can only be one operation named "TwoHeroes".
There can only be one operation named "TwoHeroes".
Validation of GraphQL query document failed
```

For 3, the ordering we get is 
`[src/main/graphql, src/demo/graphql, src/debug/graphql, src/demoDebug/graphql]`

so reversing the lookup should work in that case.